aten/src/ATen/Dimname.h

```diff
@@ -14,16 +14,23 @@ struct CAFFE2_API Dimname {
   static Dimname wildcard();
 
   NameType type() const { return type_; }
-  Symbol name() const { return name_; }
+  Symbol full_name() const { return full_name_; }
   Symbol untagged_name() const { return untagged_name_; }
 
  private:
   Dimname(Symbol name)
-    : untagged_name_(name), name_(name), type_(NameType::NORMAL) {}
-  Dimname(NameType type, Symbol name, Symbol untagged_name)
-    : untagged_name_(untagged_name), name_(name), type_(type) {}
+    : untagged_name_(name), full_name_(name), type_(NameType::NORMAL) {}
+  Dimname(NameType type, Symbol full_name, Symbol untagged_name)
+    : untagged_name_(untagged_name), full_name_(full_name), type_(type) {}
```

Comment:
is there a reason you prefer untagged_name and full_name over name and tag? To me untagged_name and full_name seem repetitive since most of the chars are in both, and then you have to manipulate a string to get the tag on its own. Is the use case just such that the isolated tag is rarely needed?

Reply:
We never need the isolated tag. It's not repetitive information because we're storing Symbols, not strings.

If we stored only `name` and `tag` it is difficult to get the `full_name` Symbol: we'd need to convert `name` to a std::string, convert `tag` to std::string, add them, and then reconstruct the Symbol. This is all very inefficient.

My hypothesis is that we need to use the `full_name` Symbol much more than the `tag` Symbol (I'm pretty sure a `tag` Symbol would never get used, but I'm not sure if if we actually need the `full_name` Symbol).

Reply:
thanks